spinalcordtoolbox/deepseg/inference.py

```diff
@@ -234,19 +234,31 @@ def segment_nnunet(path_img, tmpdir, predictor):
         img_out.change_orientation(orig_orientation)
         logger.info(f'Reorientation to original orientation {orig_orientation} done.')
 
-    # for SCI multiclass model, split the predictions into sc-seg and lesion-seg
-    if "SCI" in predictor.plans_manager.dataset_name:
-        targets = [f"_{pred_type}_seg" for pred_type in ['sc', 'lesion']]
-        fnames_out = [add_suffix(fname_prediction, target) for target in targets]
-        for i, fname_out in enumerate(fnames_out):
-            img_bin = img_out.copy()
-            img_bin.data = binarize(img_bin.data, i)
-            logger.info(f"Saving results to: {fname_out}")
-            img_bin.save(fname_out)
-    else:
+    labels = {k: v for k, v in predictor.dataset_json['labels'].items() if k != 'background'}
+    # for rootlets model (which has labels 'lvl1', 'lvl2', etc.), save the image directly without splitting
+    is_rootlet_model = all((label == f"lvl{i+1}") for i, label in enumerate(labels.keys()))
```

Comment:
Note that you can replace the `i+1` with just `i` if you use `enumerate(labels.keys(), 1)`, since [`enumerate` has a customizable start value](https://docs.python.org/3/library/functions.html#enumerate).

Reply:
Ooh, good to know!!

```suggestion
    is_rootlet_model = all((label == f"lvl{i}") for i, label in enumerate(labels.keys(), 1))
```